spec/support/thread_order_support.rb

```diff
@@ -0,0 +1,93 @@
+class ThreadOrderSupport
+  def initialize
+    @bodies  = {}
+    @threads = []
+    @queue   = [] # we may not have thread stdlib required, so may not have Queue class
```

Comment:
I'm going to have to look into this in more detail this weekend, I'm clearly confused about how it works, as I'd assume this would cause library wide checks to fail.